packages/gitmoji-changelog-markdown/src/index.js

```diff
@@ -19,7 +27,56 @@ function convert({ meta, changes }) {
     body: autolink(commit.body, meta.repository),
   }))
 
-  return generateMarkdown({ changelog })
+  return compileTemplate({ changelog })
+}
+
+function markdownFromScratch({ meta, changes }, options) {
+  const output = toMarkdown({ meta, changes })
+  return Promise.resolve(fs.writeFileSync(options.output, output))
+}
+
+function markdownIncremental({ meta, changes }, options) {
+  const { lastVersion } = meta
+
+  const currentFile = options.output
+  const tempFile = `${currentFile}.tmp`
+
+  if (!fs.existsSync(currentFile)) {
+    throw new Error(`${currentFile} doesn't exists, please execute "gitmoji-changelog init" to build it from scratch.`)
+  }
+
+  // write file for next version
+  const writer = fs.createWriteStream(tempFile, { encoding: 'utf-8' })
+  writer.write(toMarkdown({ meta, changes }))
+
+  // read original file until last tags and add it to the end
+  let lastVersionFound = false
+  return new Promise(resolve => {
+    const stream = fs.createReadStream(currentFile)
+      .pipe(es.split())
+      .pipe(es.mapSync((line) => {
+        stream.pause()
+
+        if (line.startsWith(`<a name="${lastVersion}"></a>`)) {
+          lastVersionFound = true
+        }
+        if (lastVersionFound) {
+          writer.write(`${line}\n`)
```

Comment:
you can write it in the previous condition, no ?

Reply:
No I can't, it is used to write the content of existing file after the last version. And it's a stream reading the file line by line;

For each lines of the file, 
- I'm looking for the last version (the first if)
- when it's found: I write all the next line of the file

Reply:
Thanks for the explanation 👍